### PR TITLE
Separate Python and Spark kernels

### PR DESCRIPTION
**Pull request by kyuksel**
The `Python` kernels no longer comes with Spark installed, which cuts more than 10 seconds from the kernel start-up time.

The newly added `PySpark` kernels have the former `Python` kernels' functionality.

All four kernels are displayed on the notebook UI by default.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala [hunk 1]
```diff
@@ -222,21 +223,10 @@ class NotebookInteractionSpec extends FreeSpec with LeonardoTestUtils with Befor
         val getBxPython =
           """import bx.bitset
             |bx.bitset.sys.copyright""".stripMargin
-        val sparkJob =
```
Comment (kyuksel) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala]: Moved to a newly created set below and renamed as `sparkJobToSucceed` for clarity.

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala [hunk 2]
```diff
@@ -272,7 +262,7 @@ class NotebookInteractionSpec extends FreeSpec with LeonardoTestUtils with Befor
         // http://httr.r-lib.org//index.html
 
         // it may take a little while to install
-        val installTimeout = 2 minutes
+        val installTimeout = 2.minutes
```
Comment (kyuksel) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala]: To get rid of the compiler warning...it didn't feel worth it to add a `postfixOps` import.

File: jupyter-docker/Dockerfile [hunk 3]
```diff
@@ -72,7 +72,7 @@ ENV HAILJAR hail-0.1-$HAILHASH-Spark-$SPARK_VER.jar
 ENV HAILPYTHON hail-0.1-$HAILHASH.zip
 ENV HAILZIP Hail-0.1-$HAILHASH-Spark-$SPARK_VER.zip
 ENV HAIL_HOME /etc/hail
-ENV KERNELSPEC_HOME /usr/local/share/jupyter/kernals
+ENV KERNELSPEC_HOME /usr/local/share/jupyter/kernels
```
Comment (kyuksel) [inline jupyter-docker/Dockerfile]: Given we haven't been getting failures with the typo in the file path, I suspect `KERNELSPEC_HOME` hasn't been used but couldn't be certain.

**Comment by rtitle** [comment]
Looks good! Just 1 minor comment above.

Re the discussion in https://github.com/DataBiosphere/leonardo/issues/321, could we provide a user script for @calbach which hides the pyspark kernels?

Either of these might work but we should test & bless it:
```
jupyter kernelspec remove pyspark2
jupyter kernelspec remove pyspark3
```
Alternatively:
```
rm -f ${KERNELSPEC_HOME}/pyspark2/kernel.json
rm -f ${KERNELSPEC_HOME}/pyspark3/kernel.json
```

File: jupyter-docker/Dockerfile (hunk 3 above)
Comment (kyuksel) [reply jupyter-docker/Dockerfile]: possibly because there is another `KERNELSPEC_HOME` defined in `init-actions.sh`

**Comment by kyuksel** [comment]
@rtitle, Rebased and addressed feedback

**Comment by kyuksel** [comment]
jenkins retest

**Comment by kyuksel** [comment]
jenkins retest

**Comment by kyuksel** [comment]
jenkins retest

**Pull request merged**